src/pooler.erl

```diff
@@ -220,18 +229,25 @@ take_group_member(GroupName) ->
             {_, _, X} = os:timestamp(),
             Idx = (X rem length(Pools)) + 1,
             {PoolPid, Rest} = extract_nth(Idx, Pools),
-            take_first_pool([PoolPid | Rest])
+            case {take_first_pool([PoolPid | Rest], 0), Timeout} of
```

Comment:
I also wonder if a solution like what dispcount(https://github.com/ferd/dispcount) does maybe better.  He uses an ets table with counters to determine which pool is the best to pull from.  However that has a disadvantage of doing more ets calls which would be much more expensive then just getting the timestamp ...